wrapper/VRChat.API/Client/VRChat.cs

```diff
@@ -0,0 +1,304 @@
+        public InventoryApi Inventory => _inventory;
+        public InstancesApi Instances => _instances;
+        public FavoritesApi Favorites => _favorites;
+        public MiscellaneousApi Miscellaneous => _miscellaneous;
+        public NotificationsApi Notifications => _notifications;
+        public PlayermoderationApi Moderations => _moderations;
+        public AuthenticationApi Authentication => _authentication;
+
+        public bool IsLoggedIn { get; private set; }
+
+        // Creates a new VRChatClient internally, compatible with IVRChatClient
+        internal static VRChatClient CreateInternal(Configuration configuration, string twoFactorSecret, ApiClient client, HttpClient httpClient, HttpClientHandler handler) =>
+            new VRChatClient(configuration, twoFactorSecret, client, httpClient, handler);
+
+        public List<Cookie> GetCookies()
+        {
+            return _httpClientHandler.CookieContainer?.GetAllCookies().ToList();
+        }
+
+        public async Task<bool> TryLoginAsync(CancellationToken ct = default)
+        {
+            return (await this.LoginAsync(throwOnFail: false, ct)) != null;
+        }
+
+        public async Task<CurrentUser> LoginAsync(bool throwOnFail = false, CancellationToken ct = default)
+        {
+            if (_twoFactorSecret == null)
+                throw new ArgumentNullException("This method only supports logging in with Two-Factor TOTP authentication. Please use VRChatClientBuilder.WithTwoFactorSecret() to use this method");
+
+            ApiResponse<CurrentUser> response = null;
+
+            try
+            {
+                response = await this.Authentication.GetCurrentUserWithHttpInfoAsync(cancellationToken: ct);
+
+                if (response.StatusCode == HttpStatusCode.Unauthorized)
+                {
+                    throw new UnauthorizedAccessException("401 Unauthorized");
+                }
+
+                if (response.Data.RequiresTwoFactorAuth != null && response.Data.RequiresTwoFactorAuth.Contains("totp"))
+                {
+                    var totp = new Totp(Base32Encoding.ToBytes(_twoFactorSecret));
+                    var twoFactorResponse = await this.Authentication.Verify2FAWithHttpInfoAsync(new TwoFactorAuthCode(totp.ComputeTotp()), ct);
+
+                    if (twoFactorResponse.StatusCode != HttpStatusCode.OK)
+                        throw new UnauthorizedAccessException("Failed to verify two-factor authentication with VRChat.");
+
+                    response = await this.Authentication.GetCurrentUserWithHttpInfoAsync(ct);
+                }
+                else if (response.Data.RequiresTwoFactorAuth != null)
+                    throw new InvalidOperationException("This account does not have TOTP Two Factor Authentication set up on it. This method only supports Two-Factor TOTP authentication");
+            }
+            catch
+            {
+                if (throwOnFail)
```

Comment:
`throwOnFail` is odd, just throw and let the caller decide if they want to catch
also there are many other code paths in this method that throw regardless of wheter `throwOnFail` is true or false